controllers/appointment.controller.js

```diff
@@ -0,0 +1,168 @@
+    }
+
+}
+
+/**
+ *  This api is mainly used by Doctor or Admin for:
+ *      1.to Update Status
+ *      2. to Add,update,delete appointment
+ * 
+ *  This api can also be used by patient to make changes to the appointment
+ */
+exports.updateAppointment = async (req, res) => {
+
+    try{
+        const appointment = await Appointment.findOne({
+            _id: req.params.id
+        });
+
+        if (appointment == null) {
+            return res.status(400).send({
+                message: "Appointment Record doesn't exist"
+            })
+        }
+        const user = await User.findOne({
+            userId: req.userId
+        });
+        /**
+         *  Once a appointment is created for a doctor and hospital patient cannot change doctor/hospital if he want he can cancel the appointment and create for the other doctor/hospital he want 
+         * 
+         * 
+         */
+
+        appointment.symptons = req.body.symptons != undefined ? req.body.symptons : appointment.symptons;
+        appointment.medications = req.body.medications != undefined ? req.body.medications : appointment.medications;
+        appointment.date = req.body.date != undefined ? req.body.date : appointment.date;
+       
+        /**
+         *  1. Doctor or Admin can change the status of the Appointment
+         *  2. Doctor or admin can also change , update, write prescription to the patient  
+         */ 
+        if(user.userType == constants.userType.admin || user.userType == constants.userType.doctor){
+            appointment.prescription = req.body.prescription != undefined ? req.body.prescription : appointment.prescription;
+            appointment.status = req.body.status != undefined ? req.body.status : appointment.status;
+            
+            await appointment.save();
+        }
+
+        const updatedRecord = await appointment.save();
+
+        return res.status(200).send(updatedRecord);
+    }catch(err){
+        console.log(err.message);
+        return res.status(500).send({
+            message: "Some internal error"
+        })
+    }
+}
+
+//fetching the appointment based on usertype 
+exports.getAllAppointment = async (req, res) => {
+    try {
+        const user = await User.findOne({
+            userId: req.userId
+        });
+        //admin will have access to all the appointment
+        if(user.userType == constants.userType.admin){
+            const appointment = await Appointment.find();
+            return res.status(200).send(appointment);
+        }else{
+            const appointment = await Appointment.find({userId : req.userId});
+            return res.status(200).send(appointment);
+        }
+    } catch (err) {
+        console.log(err.message);
+        return res.status(500).send({
+            message: "Some internal error"
+        })
+    }
+}
+
+//fetching appointment based on id 
+exports.getOneAppointment = async (req, res) => {
+    try{
+        const appointment = await Appointment.findOne({
```

Comment:
Validations...only the owners of the appointment should be able to call this API and get results.